### PR TITLE
Generate List go code from template

### PR DESCRIPTION
**Pull request by alban-auzeill**

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Nice!

File: uast-generator-go/generate_source.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		listTemplate.Execute(&out, ListTemplateArgs{TypeName: "*ast." + name, ListName: name + "List"})
+	}
+
+	out.WriteString("\n")
+	out.WriteString("\n// Generated Code End\n")
+
+	check(ioutil.WriteFile("goparser_generated.go", out.Bytes(), 0644))
+}
+
+func check(e error) {
```
Comment (ghost) [inline uast-generator-go/generate_source.go]: Used only once -> could be inlined

File: uast-generator-go/generate_source.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	for _, name := range []string{"Expr", "Stmt", "Decl"} {
+		listTemplate.Execute(&out, ListTemplateArgs{TypeName: "ast." + name, ListName: name + "List"})
+	}
+	for _, name := range []string{"Field", "Ident"} {
+		listTemplate.Execute(&out, ListTemplateArgs{TypeName: "*ast." + name, ListName: name + "List"})
+	}
+
+	out.WriteString("\n")
+	out.WriteString("\n// Generated Code End\n")
```
Comment (ghost) [inline uast-generator-go/generate_source.go]: Maybe it will be nicer to move this to a footer template?

**Review by ghost — APPROVED** [review]

File: uast-generator-go/generate_source.go (hunk 2 above)
Comment (alban-auzeill) [reply uast-generator-go/generate_source.go]: It was to demonstrate the two possible solutions: Template, and StringBuilder.
Go Templates seem to be powerful, but StringBuilder are easy to understand for any kind of developer.
What do you prefer?

Comment (ghost) [reply uast-generator-go/generate_source.go]: I like the template.

**Pull request merged**